src/main/java/co/cask/spanner/sink/SpannerSinkConfig.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.spanner.sink;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.spanner.common.SpannerConfig;
+
+import javax.annotation.Nullable;
+
+/**
+ * Spanner sink config
+ */
+public class SpannerSinkConfig extends SpannerConfig {
+  private static final int DEFAULT_SPANNER_WRITE_BATCH_SIZE = 100;
+
+  @Name("table")
+  @Description("Cloud Spanner table id. Uniquely identifies your table within the Cloud Spanner database")
+  @Macro
+  public String table;
+
+  @Name("batchSize")
+  @Description("Size of the batched writes to the Spanner table. " +
+    "When the number of buffered mutations is greater than this batchSize, " +
+    "the mutations are written to Spanner table, Default value is 100")
```

Comment:
how was 100 chosen? Is there a number Spanner recommends?

Reply:
Each write to Cloud Spanner contains some overhead. To maximize bulk write throughput, maximize the amount of data stored per write. A good technique is for each commit to mutate hundreds of rows. Commits with the number of mutations in the range of 1 MiB - 5 MiB rows usually provide the best performance.

Commits with more than 5 MiB mutations don't provide extra benefit, and they risk exceeding the Cloud Spanner limits on commit size and mutations per commit.

https://cloud.google.com/spanner/docs/bulk-loading

I chose hundred as a lower estimate, since they mentioned its a good technique to mutate 100s of rows.

Reply:
I see. It would be good to include that information in the doc.